packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_auth_request.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import '../../webview_flutter_platform_interface.dart';
+
+/// Defines the parameters of a pending HTTP authentication request received by
+/// the webview through a [HttpAuthRequestCallback].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class and providing a factory method that takes the [HttpAuthRequest]
+/// as a parameter.
+///
+/// {@tool sample}
+/// This example demonstrates how to extend the [HttpAuthRequest] to provide
+/// additional platform specific parameters.
+///
+/// When extending [HttpAuthRequest], additional parameters should always accept
+/// `null` or have a default value to prevent breaking changes.
+///
+/// ```dart
+/// @immutable
+/// class WKWebViewHttpAuthRequest extends HttpAuthRequest {
+///   WKWebViewHttpAuthRequest._(
+///     HttpAuthRequest authRequest,
+///     this.extraData,
+///   ) : super(
+///      onProceed: authRequest.onProceed,
+///      onCancel: authRequest.onCancel,
+///      host: authRequest.host,
+///      realm: authRequest.realm,
+///   );
+///
+///   factory WKWebViewHttpAuthRequest.fromHttpAuthRequest(
+///     HttpAuthRequest authRequest, {
+///     String? extraData,
+///   }) {
+///     return WKWebViewHttpAuthRequest._(
+///       authRequest,
+///       extraData: extraData,
+///     );
+///   }
+///
+///   final String? extraData;
+/// }
+/// ```
+/// {@end-tool}
+@immutable
+class HttpAuthRequest {
+  /// Creates a [HttpAuthRequest].
+  const HttpAuthRequest({
+    required this.onAuthenticate,
+    required this.host,
+    this.realm,
+  });
+
+  /// The callback to proceed with, or cancel an auth request.
+  ///
+  /// If `credential` is `null`, the request will be canceled.
+  final void Function(WebViewCredential? credential) onAuthenticate;
```

Comment:
I think the outcome of https://github.com/flutter/packages/pull/4140#discussion_r1320308186 was that we were going to go back to `onProceed` and `onCancel`? I'll defer to @bparrishMines on the exact callback structure though.

Reply:
Correct. I think we should separate this into  `onCancel` and `onProceed` again. And then iOS can add its additional method.